internal/job/shell/shell_test.go

```diff
@@ -316,24 +316,28 @@ func TestLockFileRetriesAndTimesOut(t *testing.T) {
 
 	lockPath := filepath.Join(dir, "my.lock")
 
-	cmd, err := acquireLockInOtherProcess(lockPath)
+	cmd, err := acquireLockInOtherProcess(t, lockPath)
 	assert.NilError(t, err)
 	defer func() { assert.NilError(t, cmd.Process.Kill()) }()
 
 	_, err = sh.LockFile(context.Background(), lockPath, 2*time.Second)
 	assert.ErrorIs(t, err, context.DeadlineExceeded)
 }
 
-func acquireLockInOtherProcess(lockfile string) (*exec.Cmd, error) {
+func acquireLockInOtherProcess(t *testing.T, lockfile string) (*exec.Cmd, error) {
+	t.Helper()
+
 	expectedLockPath := lockfile + "f" // flock-locked files are created with the suffix 'f'
 
+	t.Logf("acquiring lock in other process: %s", lockfile)
+
 	cmd := exec.Command(os.Args[0], "--", lockfile)
 	cmd.Env = []string{"TEST_MAIN_WANT_HELPER_PROCESS=1"}
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
 
 	err := cmd.Start()
-	if err != nil {
-		return cmd, err
-	}
+	assert.NilError(t, err)
```

Comment:
So this fails the test immediately if err != nil? (Looks like it... had to look it up since I'm not yet familiar with this particular assert lib). Then there would be no need to have an error return on the function.

Reply:
Yeah, good point. I've made it not return the error. If the next assertion passes, then it's a false positive. So I think it's fine to exit early.